src/DataStructures.jl

```diff
@@ -60,7 +60,7 @@ module DataStructures
     export DiBitVector
 
     export RBTree, search_node, minimum_node
-    export AVLTree
+    export AVLTree, rank
```

Comment:
The name `rank` is also used by `LinearAlgebra`.
Are these the same concept? Should we overload `LInearAlgebra.rank`?

Reply:
`rank` in Linear Algebra, means something fundamentally from what is present here.
In linear algebra,`rank` means the maximum number of linearly independent vectors in the vector space generated by the columns/rows.
Here, it correspondings to the order/position in which the keys appear in the sorted sequence.
For example, 
if the tree consists of ` keys = [2, 5, 7, 10]`, 
```
rank(2) = 1
rank(5) = 2
rank(7) = 3
rank(10) = 4
```

Reply:
ok, fair enough.
I wonder if we can name it something different to avoid a name conflict with LinearAlgebra.
Like `sort_rank` or soething